### PR TITLE
Staging V2 client side

### PR DESCRIPTION
**Pull request by cstamas**
Nexus Maven plugin and RESTLight client changes.

In short:
- restlight client got support for new resource (introduced in V2)
- nexus maven plugin uses new restlight support (V2)
- zapper transport switched from AHC to HC4 transport
- Mojo parameters cleanup
- POM cleanup (and a bit of merge problems)

### CODE REVIEW

**Comment by cstamas** [comment]
CI job:
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/339/

**Comment by cstamas** [comment]
CI passed (but is reported as failed since class UserTokenTemplateInterpolatorCustomizerTest picked up from master as this branch was rebased lacks license header).

File: nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/deploy/AbstractDeployMojo.java [hunk 1]
```diff
@@ -252,12 +313,220 @@ protected void deployStagedArtifacts()
...
+        {
+            throw new ArtifactDeploymentException( "Cannot decipher credentials for deploy!", e );
+        }
+        catch ( MalformedURLException e )
+        {
+            throw new ArtifactDeploymentException( "Malformed Nexus base URL!", e );
+        }
+    }
+
+    private String createdStagingRepositoryId;
```
Comment (nabcos) [inline nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/deploy/AbstractDeployMojo.java:401]: field declaration between methods?

**Comment by nabcos** [comment]
+1

**Comment by jdillon** [comment]
+1

**Pull request merged**